### PR TITLE
feat: add a linked-ci details modal for application and app group 

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description
Linked CI Details Modal

Fixes https://github.com/devtron-labs/devtron/issues/116

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+            <span>{environment}</span>
+            <span>{triggerMode}</span>
+            <span> {deploymentStatus} </span>
+        </div>
+    )
+}
+
+const AppList = () => {
+    return (
+        <div className="">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:40]: - remove empty className
- Use direct arrow function return
- Move to separate files

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            <div className="flexbox bcn-0 dc__uppercase pl-20 pr-20 pt-6 pb-6 dc__border-bottom dc__align-self-stretch dc__align-center dc__gap-16 fs-12 fw-6 dc__content-space">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:41]: Group similar classes, add line height. We can ditch the padding blocks and use grid-auto-row to define row height
```suggestion
            <div className="flex dc__gap-16 bcn-0 dc__uppercase pl-20 pr-20 pt-6 pb-6 dc__border-bottom fs-12 fw-6 dc__content-space">
```

Comment (eshankvaish) [reply src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:41]: The layout would not have any relationship b/w rows. PS: use grid

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                <SortableTableHeaderCell
+                    title="Application"
+                    sortOrder={SortingOrder.ASC}
+                    isSorted={false}
+                    triggerSorting={() => {}}
+                    disabled={false}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:47]: add TODO

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                />
+                <span>Deploys To (ENV)</span>
+                <span>Trigger Mode</span>
+                <span>Last Deployment Status</span>
+            </div>
+            <div className="flexbox-col bcn-0 dc__overflow-scroll">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:53]: same as above for layout

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    SortableTableHeaderCell,
+    SortingOrder,
+    getCommonSelectStyle,
+} from '@devtron-labs/devtron-fe-common-lib'
+import ReactSelect from 'react-select'
+import { LinkedCIDetailsModalProps } from './types'
+import { ReactComponent as Info } from '../../../assets/icons/ic-info-filled.svg'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+
+import AppListData from './constants'
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:16]: - camelCase
- Use service for mock, api integration

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                    isSorted={false}
+                    triggerSorting={() => {}}
+                    disabled={false}
+                />
+                <span>Deploys To (ENV)</span>
+                <span>Trigger Mode</span>
+                <span>Last Deployment Status</span>
+            </div>
+            <div className="flexbox-col bcn-0 dc__overflow-scroll">
+                {AppListData.map((appRowData) => (
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:54]:
```suggestion
                {AppListData.map((appData) => (
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+                    <AppListRow
+                        appName={appRowData.appName}
+                        environment={appRowData.environment}
+                        triggerMode={appRowData.triggerMode}
+                        deploymentStatus={appRowData.deploymentStatus}
+                    />
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:60]:
```suggestion
                    <AppListRow
                        {...appData}
                    />
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+export interface AppListRowDataType {
+    appName: string
+    environment: string
+    triggerMode: string
+    deploymentStatus: string
+}
+
+const AppListRow = ({ appName, environment, triggerMode, deploymentStatus }: AppListRowDataType) => {
+    return (
+        <div className="flexbox pl-20 pr-20 pt-8 pb-8 dc__align-self-stretch dc__align-center dc__gap-16 fs-13 fw-4 dc__content-space">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:29]: same styling issue

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            <span>{appName}</span>
+            <span>{environment}</span>
+            <span>{triggerMode}</span>
+            <span> {deploymentStatus} </span>
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:33]: ellipsis are missing

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx (hunk 8 above)
Comment (eshankvaish) [reply src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:29]: color is missing

Comment (eshankvaish) [reply src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:29]: hyperlinking is missing. We should potentially use Link component from react router dom

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+import AppListData from './constants'
+
+const commonStyles = getCommonSelectStyle()
+
+export interface AppListRowDataType {
+    appName: string
+    environment: string
+    triggerMode: string
+    deploymentStatus: string
+}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:25]: move to types

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+import ReactSelect from 'react-select'
+import { LinkedCIDetailsModalProps } from './types'
+import { ReactComponent as Info } from '../../../assets/icons/ic-info-filled.svg'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+
+import AppListData from './constants'
+
+const commonStyles = getCommonSelectStyle()
+
+export interface AppListRowDataType {
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:20]:
```suggestion
export interface LinkedCIAppList {
```

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/Pages/Shared/LinkedCIDetailsModal/types.ts [hunk 12]
```diff
@@ -0,0 +1,10 @@
+export interface LinkedCIDetailsModalProps {
+    ciPipelineName: string
+    linkedWorkflowCount: number
+}
+export interface LinkedCIAppListDto {
+    appName: string
+    environment: string
+    triggerMode: string
+    deploymentStatus: string
+}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/types.ts:10]:
```suggestion
export interface LinkedCIDetailModalProps {
    ciPipelineName: string
    linkedWorkflowCount: number
}

export interface LinkedCIAppDto {
    appName: string
    environment: string
    triggerMode: string
    deploymentStatus: string
}
```

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 13]
```diff
@@ -67,6 +68,11 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
         )
     }
 
+    handleChipClick = (e) => {
+        e.stopPropagation()
```
[the remark this answers is not in the document]
Comment (eshankvaish) [reply src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:73]: Remove the comment, and add comment for stopPropagation

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 14]
```diff
@@ -116,72 +123,82 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
...
-                            <span className="workflow-node__text-light"> {this.props.isJobCI ? 'Job' : 'Build'} </span>
-                        )}
-                        <Tippy className="default-tt" arrow placement="bottom" content={this.props.title}>
-                            <div className="dc__ellipsis-left">{this.props.title}</div>
+                    {this.props.linkedCount ? (
+                        <Tippy
+                            className="default-tt w-200"
+                            arrow={false}
+                            placement="top"
+                            content={`Build Pipeline is linked as image source in ${this.props.linkedCount} ${this.props.linkedCount === 1 ? 'workflow.' : 'workflows.'}`}
```
Comment (eshankvaish) [inline src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:140]:
```suggestion
                            content={`Build Pipeline is linked as image source in ${this.props.linkedCount} ${this.props.linkedCount === 1 ? 'workflow' : 'workflows'}.`}
```

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 15]
```diff
@@ -116,72 +123,82 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
...
-                        Select Material
-                    </button>
+                        {this.props.isCITriggerBlocked ? 'BLOCKED' : this.props.triggerType}
+                    </div>
+                    <div className="workflow-node__title flex">
+                        {/* <img src={build} className="icon-dim-24 mr-16" /> */}
+                        <div className="workflow-node__full-width-minus-Icon">
+                            {!this.props.isJobView && (
+                                <span className="workflow-node__text-light">
+                                    {this.props.isJobCI ? ' Job ' : ' Build '}
```
Comment (eshankvaish) [inline src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:163]: verify the spaces

File: src/Pages/Shared/LinkedCIDetailsModal/linkedCIAppList.scss [hunk 16]
```diff
@@ -0,0 +1,6 @@
+.linked-ci-apps {
+    display: grid;
+    grid-template-columns: 200px 200px 100px auto;
+    column-gap: 16px;
+    align-items: center;
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/linkedCIAppList.scss:5]: use utility for grid and align items center

File: src/Pages/Shared/LinkedCIDetailsModal/linkedCIAppList.scss [hunk 17]
```diff
@@ -0,0 +1,6 @@
+.linked-ci-apps {
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/linkedCIAppList.scss:1]:
```suggestion
.linked-ci-detail {
    &__table-row {
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetails.tsx [hunk 18]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import { Switch, useRouteMatch, Route } from 'react-router-dom'
+import LinkedCIDetailsModal from './LinkedCIDetailsModal'
+import { URLS } from '../../../config'
+import { LinkedCIDetailsModalProps } from './types'
+
+const LinkedCIDetails = ({ ciPipelineName, linkedWorkflowCount }: LinkedCIDetailsModalProps) => {
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetails.tsx:7]:
```suggestion
const LinkedCIDetail = (props: LinkedCIDetailsModalProps) => {
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetails.tsx [hunk 19]
```diff
@@ -0,0 +1,23 @@
...
+    const { path } = useRouteMatch()
+
+    return (
+        <Switch>
+            <Route
+                path={`${path}/${URLS.LINKED_CI_DETAILS}/:ciPipelineId`}
+                render={() => (
+                    <LinkedCIDetailsModal ciPipelineName={ciPipelineName} linkedWorkflowCount={linkedWorkflowCount} />
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetails.tsx:15]:
```suggestion
                    <LinkedCIDetailsModal {...props} />
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx [hunk 20]
```diff
@@ -0,0 +1,52 @@
...
+        </Link>
+    )
+}
+
+const LinkedCIAppList = () => {
+    const [sortOrder, setSortOrder] = useState(SortingOrder.ASC)
+
+    const handleSorting = () => {
+        setSortOrder(SortingOrder.DESC)
+    }
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx:26]: use useUrlFilter for such state

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx [hunk 21]
```diff
@@ -0,0 +1,52 @@
...
+
+    return (
+        <div>
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx:29]: add flex probably

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+import { LinkedCIDetailsModalProps } from './types'
+import { ReactComponent as Info } from '../../../assets/icons/ic-info-filled.svg'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+import LinkedCIAppList from './LinkedCIAppList'
+import './linkedCIAppList.scss'
+
+const commonStyles = getCommonSelectStyle()
+
+const LinkedCIDetailsModal = ({ ciPipelineName, linkedWorkflowCount }: LinkedCIDetailsModalProps) => {
+    return (
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:13]: - use direct return
- add default for pipelineName and count

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+        <Drawer position="right" width="800px">
+            <div className="bcn-0 h-100">
+                <div className="flex flex-justify dc__border-bottom pt-12 pr-20 pb-12 pl-20 dc__position-sticky">
+                    <h2 className="fs-16 fw-6 lh-24 m-0">{ciPipelineName}</h2>
+                    <button type="button" className="dc__transparent flex icon-dim-24" aria-label="close-modal">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:18]:
```suggestion
                    <button type="button" className="dc__transparent flex dc__no-shrink" aria-label="close-modal">
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+import './linkedCIAppList.scss'
+
+const commonStyles = getCommonSelectStyle()
+
+const LinkedCIDetailsModal = ({ ciPipelineName, linkedWorkflowCount }: LinkedCIDetailsModalProps) => {
+    return (
+        <Drawer position="right" width="800px">
+            <div className="bcn-0 h-100">
+                <div className="flex flex-justify dc__border-bottom pt-12 pr-20 pb-12 pl-20 dc__position-sticky">
+                    <h2 className="fs-16 fw-6 lh-24 m-0">{ciPipelineName}</h2>
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:17]: ellipsis handling

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+                    <button type="button" className="dc__transparent flex icon-dim-24" aria-label="close-modal">
+                        <Close className="icon-dim-24" />
+                    </button>
+                </div>
+                <div>
+                    <InfoColourBar
+                        message={`This build pipeline is linked as image source in ${linkedWorkflowCount} ${linkedWorkflowCount === 1 ? 'workflow.' : 'workflows.'}`}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:24]: common out using a function

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+                        classname="info_bar justify-start"
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:25]: check for justify-start

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+    return (
+        <Drawer position="right" width="800px">
+            <div className="bcn-0 h-100">
+                <div className="flex flex-justify dc__border-bottom pt-12 pr-20 pb-12 pl-20 dc__position-sticky">
+                    <h2 className="fs-16 fw-6 lh-24 m-0">{ciPipelineName}</h2>
+                    <button type="button" className="dc__transparent flex icon-dim-24" aria-label="close-modal">
+                        <Close className="icon-dim-24" />
+                    </button>
+                </div>
+                <div>
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:22]: flex

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+                        <div className="dc__no-shrink">
+                            {/* todo (Arun) -- fix this */}
+                            <Pagination
+                                rootClassName="flex dc__content-space pl-20 pr-20 dc__border-top"
+                                size={100}
+                                offset={0}
+                                pageSize={20}
+                                changePage={() => {}}
+                                changePageSize={() => {}}
+                            />
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:56]: make sure to add the check for DEFAULT_PAGE_SIZE

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+                                        ...commonStyles.control,
+                                        backgroundColor: 'var(--N50)',
+                                    }),
+                                }}
+                            />
+                        </div>
+                    </div>
+                    <div>
+                        <LinkedCIAppList />
+                        <div className="dc__no-shrink">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:47]: the class is redundant

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx [hunk 30]
```diff
@@ -0,0 +1,52 @@
...
+import { Link } from 'react-router-dom'
+import { SortableTableHeaderCell, SortingOrder } from '@devtron-labs/devtron-fe-common-lib'
+import { URLS } from '../../../config'
+import { LinkedCIAppListDto } from './types'
+import AppListData from './constants'
+
+const AppListRow = ({ appName, environment, triggerMode, deploymentStatus }: LinkedCIAppListDto) => {
+    return (
+        <Link to={URLS.APP}>
+            <div className="linked-ci-apps pl-20 pr-20 pt-8 pb-8 fs-13 fw-4 dc__hover-n50 ">
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx:11]: add color to header and row

**Review by eshankvaish — CHANGES_REQUESTED** [review]

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 31]
```diff
@@ -125,8 +132,13 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
                 }}
             >
                 {this.props.linkedCount ? (
-                    <Tippy className="default-tt" arrow placement="bottom" content={this.props.linkedCount}>
-                        <span className="link-count">
+                    <Tippy
+                        className="default-tt w-200"
+                        arrow={false}
+                        placement="top"
+                        content={LinkedCITippyContent(this.props.linkedCount)}
```
Comment (eshankvaish) [inline src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:139]:
```suggestion
                        content={getLinkedCITippyContent(this.props.linkedCount)}
```

File: src/Pages/Shared/LinkedCIDetailsModal/utils.ts [hunk 32]
```diff
@@ -0,0 +1,9 @@
+import { SELECT_ALL_VALUE } from '../../../config'
+
+export const parseSearchParams = (searchParams: URLSearchParams) => ({
+    environment: searchParams.get('environment') || SELECT_ALL_VALUE,
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/utils.ts:4]: move 'environment' to constants. ENVIRONMENT_FILTER_SEARCH_KEY

File: src/Pages/Shared/LinkedCIDetailsModal/utils.ts [hunk 33]
```diff
@@ -0,0 +1,9 @@
...
+})
+
+export const LinkedCITippyContent = (workflowCount: number) => {
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/utils.ts:7]: direct arrow function return

File: src/Pages/Shared/LinkedCIDetailsModal/types.ts [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+
+// Another interface to create segregation from LinkedCIAppDto
+export interface LinkedCIApp extends LinkedCIAppDto {}
+
+export type LinkedCIAppListFilterParams = BaseFilterQueryParams<SortableKeys> & {
+    environment: string
+}
+
+export type CIPpelineEnviromentList = string[]
+export interface LinkedCIAppListProps {
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/types.ts:31]:
```suggestion
export type CIPipelineEnviromentList = string[]

export interface LinkedCIAppListProps {
```

File: src/Pages/Shared/LinkedCIDetailsModal/index.ts [hunk 35]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line import/prefer-default-export
+export { default as LinkedCIDetail } from './LinkedCIDetail'
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/index.ts:2]: check if the comment can be removed after back merge from main

File: src/Pages/Shared/LinkedCIDetailsModal/constants.ts [hunk 36]
```diff
@@ -0,0 +1,19 @@
+import { LinkedCIApp } from './types'
+import { DEFAULT_SHIMMER_LOADING_TABLE_ROWS, SELECT_ALL_VALUE } from '../../../config'
+
+export const appListLoading: LinkedCIApp[] = Array.from(Array(DEFAULT_SHIMMER_LOADING_TABLE_ROWS).keys()).map(
+    (index) => ({
+        appId: index,
+        appName: '',
+        deploymentStatus: 'succeeded',
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/constants.ts:8]: use from the object

File: src/Pages/Shared/LinkedCIDetailsModal/constants.ts [hunk 37]
```diff
@@ -0,0 +1,19 @@
...
+        environmentName: '',
+        triggerMode: '',
+    }),
+)
+
+export enum SortableKeys {
+    appName = 'app_name',
+}
+
+export const DEFAULT_ALL_ENV = { label: 'All Environments', value: SELECT_ALL_VALUE }
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/constants.ts:19]:
```suggestion
export const ALL_ENVIRONMENTS_OPTION = { label: 'All Environments', value: SELECT_ALL_VALUE }
```

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 38]
```diff
@@ -0,0 +1,215 @@
...
+import { getAppList, getLinkedCIPipelineEnvironmentList } from './service'
+import { LinkedCITippyContent, parseSearchParams } from './utils'
+import { API_STATUS_CODES, SELECT_ALL_VALUE } from '../../../config'
+import { NodeAttr } from '../../../components/app/details/triggerView/types'
+import { DEFAULT_ALL_ENV, SortableKeys } from './constants'
+import { preventBodyScroll } from '../../../components/common'
+
+const LinkedCIDetailsModal = ({ handleClose, workflows }: LinkedCIDetailModalProps) => {
+    const { ciPipelineId } = useParams<{ ciPipelineId: string }>()
+    const commonStyles = getCommonSelectStyle()
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:35]: move to constants

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 39]
```diff
@@ -0,0 +1,215 @@
...
+            environment: environmentOption.value,
+        })
+    }
+
+    const selectOptions = [DEFAULT_ALL_ENV, ...(envList ?? []).map((env) => ({ label: env, value: env }))]
+
+    const selectedOption = selectOptions.find((option) => option.value === environment) ?? {
+        label: 'All Environments',
+        value: SELECT_ALL_VALUE,
+    }
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:81]: use from constant

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetail.tsx [hunk 40]
```diff
@@ -0,0 +1,21 @@
...
+const LinkedCIDetail = (props: LinkedCIDetailModalProps) => {
+    const { path } = useRouteMatch()
+
+    return (
+        <Switch>
+            <Route
+                path={`${path}/${URLS.LINKED_CI_DETAILS}/:ciPipelineId`}
+                render={() => <LinkedCIDetailsModal {...props} />}
+                exact
+            />
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetail.tsx:16]: check for children vs render

Comment (eshankvaish) [reply src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetail.tsx:16]: wrap with drawer here

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 41]
```diff
@@ -0,0 +1,215 @@
...
+                                        ...commonStyles.control(base, state),
+                                        width: 200,
+                                        height: 32,
+                                        minHeight: 32,
+                                    }),
+                                    menu: (base) => ({
+                                        ...base,
+                                        zIndex: 5,
+                                    }),
+                                }}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:181]: move to constants. environmentFilterDropdownStyles

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx [hunk 42]
```diff
@@ -0,0 +1,215 @@
...
+                {!showLoadingState && result.totalCount > DEFAULT_BASE_PAGE_SIZE ? (
+                    <Pagination
+                        rootClassName="flex dc__content-space pl-20 pr-20 dc__border-top dc__no-shrink"
+                        size={result.totalCount}
+                        offset={offset}
+                        pageSize={pageSize}
+                        changePage={changePage}
+                        changePageSize={changePageSize}
+                    />
+                ) : null}
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIDetailsModal.tsx:209]: use &&

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx [hunk 43]
```diff
@@ -0,0 +1,106 @@
...
+import { Link } from 'react-router-dom'
+import { SortableTableHeaderCell, AppStatus, GenericEmptyState } from '@devtron-labs/devtron-fe-common-lib'
+import { URLS } from '../../../config'
+import { LinkedCIApp, LinkedCIAppListProps } from './types'
+import EmptyStateImage from '../../../assets/img/empty-noresult@2x.png'
+import { StatusConstants } from '../../../components/app/list-new/Constants'
+import { SortableKeys, appListLoading } from './constants'
+
+const AppListRow = ({ appId, appName, deploymentStatus, environmentName, triggerMode, environmentId }: LinkedCIApp) => {
+    return (
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx:11]: direct arrow function return

File: src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx [hunk 44]
```diff
@@ -0,0 +1,106 @@
...
+        </Link>
+    )
+}
+
+const LinkedCIAppList = ({ appList, totalCount, isLoading, urlFilters }: LinkedCIAppListProps) => {
+    const renderClearFilterButton = () => (
+        <button type="button" onClick={urlFilters.clearFilters} className="cta secondary flex h-32">
+            Clear Filters
+        </button>
+    )
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/LinkedCIAppList.tsx:45]: move inside the if

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by eshankvaish — DISMISSED** [review]

File: src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx [hunk 45]
```diff
@@ -2225,6 +2226,10 @@ export default function EnvTriggerView({ filteredAppIds, isVirtualEnv }: AppGrou
     }
 
     const renderWorkflow = (): JSX.Element => {
+        const handleModalClose = () => {
```
Comment (AbhishekA1509) [inline src/components/ApplicationGroup/Details/TriggerView/EnvTriggerView.tsx:2229]: Its anti pattern to have functions inside function

File: src/components/app/details/triggerView/TriggerView.tsx [hunk 46]
```diff
@@ -1314,6 +1315,10 @@ class TriggerView extends Component<TriggerViewProps, TriggerViewState> {
     }
 
     renderWorkflow() {
+        const handleModalClose = () => {
```
Comment (AbhishekA1509) [inline src/components/app/details/triggerView/TriggerView.tsx:1318]: Same here

**Review by eshankvaish — DISMISSED** [review]

File: src/Pages/Shared/LinkedCIDetailsModal/utils.ts [hunk 47]
```diff
@@ -0,0 +1,16 @@
+import { SELECT_ALL_VALUE, URLS } from '../../../config'
+import { ENVIRONMENT_FILTER_SEARCH_KEY } from './constants'
+import { LinkedCIAppURLProps as LinkedCIAppUrlProps } from './types'
```
Comment (eshankvaish) [inline src/Pages/Shared/LinkedCIDetailsModal/utils.ts:3]: fix the issue

File: src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx [hunk 48]
```diff
@@ -125,11 +132,16 @@ export class TriggerCINode extends Component<TriggerCINodeProps> {
...
                 {this.props.linkedCount ? (
-                    <Tippy className="default-tt" arrow placement="bottom" content={this.props.linkedCount}>
-                        <span className="link-count">
+                    <Tippy
+                        className="default-tt w-200"
+                        arrow={false}
+                        placement="top"
+                        content={getLinkedCITippyContent(this.props.linkedCount)}
+                    >
+                        <button type="button" className="link-count cursor dc__hover-border-n300" onClick={this.handleLinkedCIWorkflowChipClick}>
```
Comment (eshankvaish) [inline src/components/app/details/triggerView/workflow/nodes/triggerCINode.tsx:141]: verify the styling for the button

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by eshankvaish — APPROVED** [review]
LGTM

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**